File: LeetCode/java/434_Number_of_Segments_in_a_String.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int countSegments(String s) {
        List<String> list = new ArrayList<>();

        StringBuilder temp = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') {
                if (temp.length() == 0) {
                    continue;
                }

                list.add(temp.toString());
                temp.setLength(0);
            }
            else {
                temp.append(s.charAt(i));
            }
        }

        if (temp.length() != 0) {
            list.add(temp.toString());
        }

        return list.size();
    }
}

File: LeetCode/cpp/434_Number_of_Segments_in_a_String.cpp
class Solution {
public:
	int countSegments(string s) {
		vector<string> v;

		string temp;
		for (int i = 0; i < s.size(); i++) {
			if (s[i] == ' ') {
				if (temp.size() == 0) {
					continue;
				}

				v.push_back(temp);
				temp.clear();
			}
			else {
				temp += s[i];
			}
		}

		if (!temp.empty()) {
			v.push_back(temp);
		}

		return v.size();
	}
};